File: app/src/main/java/kh/edu/rupp/to_dolistapp/adapters/UserViewPagerAdapter.java
package kh.edu.rupp.to_dolistapp.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import kh.edu.rupp.to_dolistapp.views.AllFragment;
import kh.edu.rupp.to_dolistapp.views.CachedFragment;
import kh.edu.rupp.to_dolistapp.views.LocalFragment;

public class UserViewPagerAdapter extends FragmentStateAdapter {
    public UserViewPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position){
            case 0:
                return new AllFragment();
            case 1:
                return new CachedFragment();
            case 2:
                return new LocalFragment();
            default:
                return new AllFragment();
        }
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: app/src/main/java/kh/edu/rupp/to_dolistapp/adapters/UserViewPagerAdapter.kt
package kh.edu.rupp.to_dolistapp.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import kh.edu.rupp.to_dolistapp.views.AllFragment
import kh.edu.rupp.to_dolistapp.views.CachedFragment
import kh.edu.rupp.to_dolistapp.views.LocalFragment

class UserViewPagerAdapter(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        when (position) {
            0 -> return AllFragment()
            1 -> return CachedFragment()
            2 -> return LocalFragment()
            else -> return AllFragment()
        }
    }

    override fun getItemCount(): Int {
        return 3
    }
}
